pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -342,9 +405,36 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 		}
 	}
 	if cpuset, ok := s.GetCPUSet(string(pod.UID), container.Name); ok {
-		p.updateCPUsToReuse(pod, container, cpuset)
-		klog.InfoS("Static policy: container already present in state, skipping", "pod", klog.KObj(pod), "containerName", container.Name)
-		return nil
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			klog.InfoS("Static policy: container already present in state, attempting InPlacePodVerticalScaling", "pod", klog.KObj(pod), "containerName", container.Name)
+			cpusInUse := getAssignedCPUsOfSiblings(s, string(pod.UID), container.Name)
+			s.Delete(string(pod.UID), container.Name)
+			toRelease := cpuset.Difference(cpusInUse)
+			previousdefaultcpuset := s.GetDefaultCPUSet()
+			s.SetDefaultCPUSet(s.GetDefaultCPUSet().Union(toRelease))
+			p.updateCPUsToReuse(pod, container, cpuset)
+
+			// Call Topology Manager to get the aligned socket affinity across all hint providers.
+			hint := p.affinity.GetAffinity(string(pod.UID), container.Name)
```

Comment:
what guarantees we have a valid hint at this point?
In the admission flow the topology manager first gather hints from all the providers, then merges and selects the best according to the configured policy, finally stores the best hint and asks all the providers to allocate using this hint. Now: I don't recall on top of my head if we clear the hints after the admission is completed or if we happen to keep them lingering (which could be a bug!), but surely these hints can be obsolete, leading to wrong or suboptimal allocations.

Reply:
ok answering myself this is handled again retrying the admission chain in `canResizePod`

Reply:
In the process of understanding the flow, I created this, perhaps it will be helpful.
![vpa_cpu_static](https://github.com/kubernetes/kubernetes/assets/159060890/acb70113-c8ab-47c3-b31f-fba07c80a219)

Reply:
Do we risk races between the in-place resize - which does admission checks - and the admission check flows?

Reply:
I believe not, my understanding is that calls are sequential and in-place resize is protected with podResizeMutex.
https://github.com/kubernetes/kubernetes/blob/546f7c30860dcdecb75c544230a1b7cdf5bd5958/pkg/kubelet/kubelet.go#L2841

Reply:
yes, resize requests are mutually exclusive between each other, and so are admission requests between each other. So both flows are safe in isolation. But what about the interaction between the two? can a regular admission race with an admission done in the resize? that's likely a non-issue for non-exclusive cpu allocation but can be an issue with exclusive cpu allocation. Let's check we are race-free.

Reply:
I believe we are safe since SyncPod is handling both cases. This corner case I cannot  test with the unit tests i have today in this PR, i believe.  https://github.com/kubernetes/kubernetes/pull/124296 , must be merged first to be able to create in this PR the new add on test cases afterwards.